pkg/reconciler/v1alpha1/autoscaling/kpa/kpa_scaler.go

```diff
@@ -87,16 +86,14 @@ func (ks *kpaScaler) getAutoscalerConfig() *autoscaler.Config {
 }
 
 // pre: 0 <= min <= max && 0 <= x
-func applyBounds(min, max int32) func(int32) int32 {
-	return func(x int32) int32 {
-		if x < min {
-			return min
-		}
-		if max != 0 && x > max {
-			return max
-		}
-		return x
+func applyBounds(min, max, x int32) int32 {
+	if x < min {
+		return min
+	}
+	if max != 0 && x > max {
```

Comment:
Why do we not apply the max bound when max == 0?

It is initially unclear to me and may deserve a comment.

Reply:
if you look at `pa.ScaleBounds` it says that `0` means infinite.